vowpalwabbit/chained_proxy_iterator.h

```diff
@@ -52,39 +52,53 @@ struct chained_proxy_iterator
   {
     ++_current;
     // TODO: don't rely on audit_end
-    if (_current == (*_outer_current).audit_end() && (_outer_current != _outer_end))
+    if (_current == (*_outer_current).feats.audit_end() && (_outer_current != _outer_end))
     {
       ++_outer_current;
-      _current = (*_outer_current).audit_begin();
+      _current = (*_outer_current).feats.audit_begin();
     }
     return *this;
   }
 
-  // TODO jump full feature groups.
+  // UB if diff < 0
   chained_proxy_iterator& operator+=(difference_type diff)
   {
-    for (size_t i = 0; i < diff; i++) { operator++(); }
-    return *this;
+    assert(diff >= 0);
+    while (true)
```

Comment:
I am afraid of `while (true)` loops since some one at some point might forget to add an exit somewhere. Is there maybe a way of limiting this to something specific?

Reply:
While true is a bit scary, I'm not sure what I should change this to though and still feel confident in the condition

Reply:
what about ! the first if statement?

Reply:
Resolved